### PR TITLE
[LayoutTests] Rebaseline text tests with pixel refs

### PR DESCRIPTION
**Pull request by JonWBedard**
#### 67855f6114ec883e030983a1f469697fbd52cef2
<pre>
[LayoutTests] Rebaseline text tests with pixel refs
<a href="https://rdar.apple.com/127396069">rdar://127396069</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=273595">https://bugs.webkit.org/show_bug.cgi?id=273595</a>

Reviewed by Sam Sneddon.

There are a number of text based tests which have pixel refs. Since these tests explicitly
dumpText, they cannot also be pixel tests.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-break/relpos-inline-hit-testing-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-break/relpos-inline-hit-testing-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-flexbox/intrinsic-size/col-wrap-004-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-flexbox/intrinsic-size/col-wrap-004-expected.xht: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-flexbox/intrinsic-size/col-wrap-005-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-flexbox/intrinsic-size/col-wrap-005-expected.xht: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-001-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-001-expected.xht: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-002-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-002-expected.xht: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-003-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/colspan-003-expected.xht: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/percent-height-overflow-auto-in-restricted-block-size-cell-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-tables/percent-height-overflow-auto-in-restricted-block-size-cell-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-004-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/width-height-004-expected.txt: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/278402@main">https://commits.webkit.org/278402@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/b07c0a1cc6242720bdfdda7098cc8885567ed2e7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50265 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29558 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2562 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53524 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/955 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35786 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/595 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41012 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52364 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27222 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43265 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22111 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/24638 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/518 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8643 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/46624 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/582 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55109 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25361 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/515 "Found 1 new test failure: imported/w3c/web-platform-tests/html/browsers/the-window-object/open-close/open-features-tokenization-noreferrer.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48413 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/26624 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43449 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47451 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11057 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27486 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26354 "Built successfully") | | |

### CODE REVIEW

File: LayoutTests/resources/testharnessreport.js [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 // Setup for WebKit JavaScript tests
 if (self.testRunner) {
-    testRunner.dumpAsText();
+    testRunner.dumpAsText(testRunner.shouldDumpPixels());
```
Comment (Ahmad-S792) [inline LayoutTests/resources/testharnessreport.js]: Just for my knowledge asking - LayoutTest use this but WPT in imported folder use `LayoutTests/imported/w3c/web-platform-tests/resources/testharnessreport.js`?

Comment (JonWBedard) [reply LayoutTests/resources/testharnessreport.js]: No, the WPT tests use this too when run under `run-webkit-tests`, it's how WPT tests know how to interact with `WebKitTestRunner`.

**Comment by gsnedders** [comment]
So it turns out that upstream WPT doesn't lint for this: https://github.com/web-platform-tests/wpt/blob/c5ced76d2189c00405d2c153f406fdcd8f91781d/tools/lint/lint.py#L451-L469

This is another thing we should fix, and just fix the tests.

**Comment by JonWBedard** [comment]
Clearly, this doesn't work, `testRunner.shouldDumpPixels()` isn't a thing. I'll be working on a solution.

**Comment by gsnedders** [comment]
There's a number of moving parts here, and a number of places where we could fix this.

 * The importer is (was?) copying the `<link rel=match>` files to `-expected.html` (I suspect https://commits.webkit.org/277673@main may well have fixed that, and they will no longer be copied, as like the upstream code it won't it as a reftest any more).
 * Fixing the tests to avoid both being defined (and probably add a lint, https://github.com/web-platform-tests/wpt/issues/46058).
 * Once we move over to actually paying attention to `<link rel=match>` instead of just looking for `-expected.html`, making sure we don't load that URL in this case.

**Review by gsnedders — APPROVED** [review]
This seems fine; we should have a longer-term solution to avoid these being added to WPT in the first place, and make sure we haven't missed anything from this PR.

**Pull request merged**